builtin/mainmenu/settings/shader_component.lua

```diff
@@ -0,0 +1,160 @@
+--Minetest
+--Copyright (C) 2022 rubenwardy
+--
+--This program is free software; you can redistribute it and/or modify
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+local shadow_levels_labels = {
+	fgettext("Disabled"),
+	fgettext("Very Low"),
+	fgettext("Low"),
+	fgettext("Medium"),
+	fgettext("High"),
+	fgettext("Very High")
+}
+
+
+local shadow_levels_options = {
+	table.concat(shadow_levels_labels, ","),
+	{ "0", "1", "2", "3", "4", "5" }
+}
+
+
+local function get_shadow_mapping_id()
+	local shadow_setting = core.settings:get("shadow_levels")
+	for i = 1, #shadow_levels_options[2] do
+		if shadow_setting == shadow_levels_options[2][i] then
+			return i
+		end
+	end
+	return 1
```

Comment:
```suggestion
	local level = tonumber(core.settings:get("shadow_levels"))
	if level and level >= 0 and level < #shadow_levels_labels then
		return level + 1
	end
	return 1
```
I'm pretty sure this is equivalent--I don't see why we need an array of stringified numbers.